### PR TITLE
feat(primitives, stages): improve checkpoint logs readability

### PR DESCRIPTION
**Pull request by shekhirin**
```console
2023-10-26T15:21:35.399539Z  INFO reth::node::events: Executing stage pipeline_stages=1/13 stage=Headers from=18427744 checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.399816Z  INFO execute{stage=Headers}: sync::stages::headers: Target block already reached checkpoint=18427744 target=Hash(0x43d6df2b4326aa863c4fb186ec07cfc3ea728a3405a7ed6c2da4e7c3d086b186)
2023-10-26T15:21:35.400127Z  INFO reth::node::events: Stage finished executing pipeline_stages=1/13 stage=Headers checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.400417Z  INFO reth::node::events: Executing stage pipeline_stages=2/13 stage=TotalDifficulty from=18427744 checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.400455Z  INFO reth::node::events: Stage finished executing pipeline_stages=2/13 stage=TotalDifficulty checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.400713Z  INFO reth::node::events: Executing stage pipeline_stages=3/13 stage=Bodies from=18427744 checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.400781Z  INFO reth::node::events: Stage finished executing pipeline_stages=3/13 stage=Bodies checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.401084Z  INFO reth::node::events: Executing stage pipeline_stages=4/13 stage=SenderRecovery from=18427744 checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.401130Z  INFO reth::node::events: Stage finished executing pipeline_stages=4/13 stage=SenderRecovery checkpoint=18427744 progress=100.0% eta=unknown
2023-10-26T15:21:35.401407Z  INFO reth::node::events: Executing stage pipeline_stages=5/13 stage=Execution from=14466361 checkpoint=14466361 progress=62.8% eta=unknown
2023-10-26T15:21:38.387337Z  INFO reth::cli: Status connected_peers=2 stage=Execution checkpoint=14466361 progress=62.8% eta=unknown
```

### CODE REVIEW

File: bin/reth/src/node/events.rs [hunk 1]
```diff
@@ -255,14 +274,26 @@ where
...
+                        checkpoint = %this.state.current_checkpoint.block_number,
+                        %progress,
+                        eta = %this.state.eta.fmt_for_stage(stage_id),
+                        "Status"
+                    );
+                } else {
+                    info!(
+                        target: "reth::cli",
+                        connected_peers = this.state.num_connected_peers(),
+                        stage = %stage_id.to_string(),
```
Comment (mattsse) [inline bin/reth/src/node/events.rs:291]: do we need to_string here?

File: crates/stages/src/pipeline/event.rs [hunk 2]
```diff
@@ -12,21 +13,17 @@ use reth_primitives::stage::{StageCheckpoint, StageId};
...
         checkpoint: Option<StageCheckpoint>,
     },
     /// Emitted when a stage has run a single time.
     Ran {
-        /// 1-indexed ID of the stage that was run out of total stages in the pipeline.
-        pipeline_position: usize,
-        /// Total number of stages in the pipeline.
-        pipeline_total: usize,
+        /// Pipeline stages progress.
+        pipeline_stages_progress: PipelineStagesProgress,
```
Comment (mattsse) [inline crates/stages/src/pipeline/event.rs:26]: I like this

**Review by mattsse — APPROVED** [review]
lgtm

File: bin/reth/src/node/events.rs (hunk 1 above)
Comment (shekhirin) [reply bin/reth/src/node/events.rs]: we don't, fixed

**Pull request merged**